jwst/tests_nightly/general/miri/test_miri_steps_single.py

```diff
@@ -283,3 +282,97 @@ def test_miri_setpointing(self):
         outputs = [(input_file,
                     'jw80600010001_02101_00001_mirimage_uncal_ref.fits')]
         self.compare_outputs(outputs)
+
+
+@pytest.mark.bigdata
+class TestMIRIMasterBackground_LRS(BaseJWSTTest):
+    input_loc = 'miri'
+    ref_loc = ['test_masterbackground', 'lrs', 'truth']
+    test_dir = ['test_masterbackground', 'lrs']
+
+    rtol = 0.000001
+
+    def test_miri_masterbackground_lrs_user1d(self):
+        """
+        Regression test of masterbackgound subtraction with lrs, with user provided 1-D background
+        """
+
+        # input file has the background added
+        # Copy original version of file to test file, which will get overwritten by test
+        input_file = self.get_data(*self.test_dir,
+                                   'miri_lrs_sci+bkg_cal.fits',
+                                   docopy=True  # always produce local copy
+                                   )
+
+        # user provided 1-D background
+        input_1d_bkg_file = self.get_data(*self.test_dir,
+                                         'miri_lrs_bkg_x1d.fits',
+                                          docopy=True)
+
+        result = MasterBackgroundStep.call(input_file,
+                                           user_background=input_1d_bkg_file,
+                                           save_results=True)
+        # _____________________________________________________________________
+        # Test 1
+        # Run extract1D on the master background subtracted data (result)  and
+        # the science data with not background added
+
+        # run 1-D extract on results from MasterBackground step
+        result_1d = Extract1dStep.call(result, save_results=True)
+
+        # run 1-D extract on original science data without background
+        input_sci_cal_file = self.get_data(*self.test_dir,
+                                            'miri_lrs_sci_cal.fits')
+        # find the 1D extraction of this file
+        # find Extract1dStep on sci data to use the same version of
+        # this rountine run on both files  rather than running it off line
+        # and having the 1-D extracted science file stored as input to step
+
+        sci_cal_1d = Extract1dStep.call(input_sci_cal_file, save_results=True)
```

Comment:
I don't think you need to specify save_results=True.

Reply:
fixed